packages/pulumi-aws/src/apps/storage/StorageApp.ts

```diff
@@ -0,0 +1,77 @@
+import {
+    defineApp,
+    createGenericApplication,
+    ApplicationContext,
+    ApplicationHooks,
+    ApplicationConfig
+} from "@webiny/pulumi-sdk";
+
+import { createCognitoResources } from "./StorageCognito";
+import { createDynamoTable } from "./StorageDynamo";
+import { createFileManagerBucket } from "./StorageFileManager";
+
+export interface StorageAppConfig extends Partial<ApplicationHooks> {
+    protect?(ctx: ApplicationContext): boolean;
+    legacy?(ctx: ApplicationContext): StorageAppLegacyConfig;
+}
+
+export interface StorageAppLegacyConfig {
+    useEmailAsUsername?: boolean;
+}
+
+export const StorageApp = defineApp({
+    name: "storage",
+    config(app, config: StorageAppConfig) {
+        const protect = config.protect?.(app.ctx) ?? app.ctx.env !== "dev";
+        const legacyConfig = config?.legacy?.(app.ctx) ?? {};
+
+        // Setup DynamoDB table
+        const table = createDynamoTable(app, { protect });
+
+        // Setup Cognito
+        const cognito = createCognitoResources(app, {
+            protect,
+            useEmailAsUsername: legacyConfig.useEmailAsUsername ?? false
+        });
+
+        // Setup file storage bucket
+        const fileManagerBucket = createFileManagerBucket(app, { protect });
+
+        app.addOutputs({
+            fileManagerBucketId: fileManagerBucket.output.id,
```

Comment:
I can see we have DynamoDB part covered here. BTW what's going on with the ElasticSearch DB?

Remember - we have two Webiny setups: DynamoDB-only and DynamoDB+ElasticSearch. Do we have both setups covered here or are we missing the latter one?

Reply:
Right, elastic was not covered yet. I think I need to do it in a separate PR, because it's quite a big task.

Reply:
What was the approach you decided to take?
Maybe it would be better if we would jump on a call on Monday?